packages/svelte/src/compiler/phases/3-transform/client/visitors/template.js

```diff
@@ -1753,6 +1753,8 @@ export const template_visitors = {
 		// push into init, so that bindings run afterwards, which might trigger another run and override hydration
 		context.state.init.push(
 			b.stmt(
+				// Even if the {@html } block seems static, it should be preserved as such, because it could be dynamic originally
+				// (like {@html browser ? 'foo' : 'bar'} which is then different between client and server.
```

Comment:
not sure I understand what this is getting at?

Reply:
I moved that comment from the test to this location to remind our future selves to never optimize this. Does this make it clearer: ?
```suggestion
				// Always call the `$.html ` function, even in a potential future where we could optimize this away for static content:
				// Even if the {@html } block seems static, it should be preserved as such, because it could be dynamic originally
				// (like {@html browser ? 'foo' : 'bar'} which is then different between client and server.
```

Reply:
I feel like this is a bit superfluous — we already have tests that would fail if we changed the behaviour, which feels like a more reliable defense than a comment

Reply:
We don't have tests for this currently, that's the thing. The test setup get changed a while ago to just pass props that can be different between server and client, but that means `@html` needs to be dynamic, in which case it obviously isn't removed.
I guess I need to enhance the test suite to add back that capability ...